chunky/src/java/se/llbit/math/Octree.java

```diff
@@ -490,19 +502,33 @@ public boolean enterBlock(Scene scene, Ray ray, BlockPalette palette) {
       ray.setCurrentMaterial(currentBlock);
 
       if (currentBlock.localIntersect) {
+        // Other functions expect the ray origin to be in the block they test so here time
+        // to update it
+        ray.o.scaleAdd(distance, ray.d);
```

Comment:
Should we add a comment on the fact that `offsetX,Y,Z` need to be updated but this is deferred to lines 515 and 523?
It's clear when you know what the code _should_ do, though.